spec/operators/every-spec.ts

```diff
@@ -1,274 +1,365 @@
-    const sourceSubs = '^      !          ';
-    const expected =   '--------          ';
-    const unsub =      '       !          ';
-
-    const result = source.pipe(
-      mergeMap((x: any) => of(x)),
-      every(predicate),
-      mergeMap((x: any) => of(x))
-    );
-
-    expectObservable(result, unsub).toBe(expected);
-    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  it('should not break unsubscription chains when result observable is unsubscribed', () => {
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = { a: 5, b: 10, c: 15 };
+      const e1 = hot('  --a--b--c--d--e--|', values);
+      const e1subs = '  ^------!          ';
+      const expected = '--------          ';
+      const unsub = '   -------!          ';
+
+      const result = e1.pipe(
+        mergeMap((x: any) => of(x)),
+        every(predicate),
+        mergeMap((x: any) => of(x))
+      );
+
+      expectObservable(result, unsub).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
   it('should propagate error if predicate eventually throws', () => {
-    const source = hot('--a--b--c--d--e--|');
-    const sourceSubs = '^       !';
-    const expected =   '--------#';
-
-    function faultyPredicate(x: string) {
-      if (x === 'c') {
-        throw 'error';
-      } else {
-        return true;
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('  --a--b--c--d--e--|');
+      const e1subs = '  ^-------!';
+      const expected = '--------#';
+
+      function faultyPredicate(x: string) {
+        if (x === 'c') {
+          throw 'error';
+        } else {
+          return true;
+        }
       }
-    }
 
-    expectObservable(source.pipe(every(faultyPredicate))).toBe(expected);
-    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+      expectObservable(e1.pipe(every(faultyPredicate))).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
-  it('should emit true if single source element match with predicate', () => {
-    const source = hot('--a--|', {a: 5});
-    const sourceSubs = '^    !';
-    const expected =   '-----(x|)';
+  it('should emit true if single source element matches with predicate', () => {
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = { a: 5 };
+      const e1 = hot('  --a--|   ', values);
+      const e1subs = '  ^----!   ';
+      const expected = '-----(x|)';
 
-    expectObservable(source.pipe(every(predicate))).toBe(expected, {x: true});
-    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+      expectObservable(e1.pipe(every(predicate))).toBe(expected, { x: true });
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
-  it('should emit true if Scalar source matches with predicate', () => {
-    const source = of(5);
```

Comment:
issue: Probably best to not change any sources that are currently declared using `of()` to `cold()`, as there is a subtle difference - see #5760 and its associated issue for background.

Is this something that you've done in other PRs? I'm wondering whether or not I missed it in any of the other reviews.

Reply:
OK, I will change this, but I have been changing this in my other PRs as well. Should I revert from other tests?

Reply:
Oh, didn't see that you already asked... Yes, I was changing in other PRs as well. I'll make a revert PR.

Reply:
@cartant I started reading #5760 and I found it to be quite large to me. I'll give it try next time, but for now, out of curiosity, I'd like to know what would e.g. `cold('a')` represent? This:
`new Observable(subscriber => subscriber.next('a'))`
or this:
`new Observable(subscriber => setTimeout(() => subscriber.next('a'), 0))`
Is #5760 trying change behavior from one to another maybe?

Reply:
`cold('a')` essentially represents `new Observable(subscriber => asapScheduler.schedule(() => subscriber.next('a')))`.

Reply:
Updated PR, I think I reverted everything. Other reverts coming...
I didn't know about this subtle distinction. Thanks a lot for explanations Nicholas. Any chance to explain why `asapScheduler` is used for completion notifications? If easier, you can post a link to an issue or to a source code.
Is completion notification scheduled on `asapScheduler` only for frame 0 or for every frame?

Reply:
Thanks for the update.

Actually, what I said above was wrong. [All notifications are emitted on the test scheduler](https://github.com/ReactiveX/rxjs/blob/1f021fe280e707a6bfbd36dcc8880d17c3e64fd2/src/internal/testing/ColdObservable.ts#L35-L50), so it's kinda equivalent to:

```ts
new Observable(subscriber => {
  asapScheduler.schedule(() => subscriber.next('a'));
  asapScheduler.schedule(() => subscriber.complete());
})
```